data/data/aws/vpc/sg-master.tf

```diff
@@ -148,6 +148,46 @@ resource "aws_security_group_rule" "master_ingress_kubelet_insecure" {
   self      = true
 }
 
+resource "aws_security_group_rule" "master_ingress_kube_scheduler_from_worker" {
```

Comment:
Do workers need direct access to the scheduler?  This is surprising to me, but I'm not familiar with the scheduler implementation.

Reply:
(cluster-monitoring) prometheus is running on workers, and prometheus needs to scrape the metrics

Reply:
> ... and prometheus needs to scrape the metrics

Ugh.  Ok.  But it really feels like we should get Prometheus and other monitoring out into a different subnet or something then.  Having these ports and the 9000-9999 range on all machines open to all workers seems like one more thing to worry about for security.

Reply:
These services are already protected, we need to do a better job of scanning them.

We should not be depending on security groups for security.  We need to find a way to ensure teams can own both exposing their metrics and securing their metrics.  In the core control plane case we've already done that, but we need to keep improving.

Reply:
> We should not be depending on security groups for security.

No, but the more barriers there are between a given bug an a useful exploit, the better.

Reply:
Agreed. My understanding is that once the 1.12 rebase lands the masters team will change the secure port to something in the open range (9000-10000).

Reply:
Though the default secure ports are 10259 for the scheduler, and 10257 for the controller-manager. As we don't change this for the kubelet, I'd be hesitant to deviate from the defaults for other components.